app/controllers/application_controller.rb

```diff
@@ -36,4 +37,8 @@ class ApplicationController < ActionController::Base
   protected def attach_ip_in_header
     @client_ip = request.ip
   end
+
+  protected def setup_locale(&action)
+    I18n.with_locale(User.languages[@logged_in_user.try(:language) || :en], &action)
```

Comment:
you should be saving 'en' in language preference why saving as  English?
it should be something like, I18n.with_locale(current_user.preference)